astropy/timeseries/periodograms/lombscargle/implementations/main.py

```diff
@@ -84,7 +85,7 @@ def validate_method(method, dy, fit_mean, nterms, frequency, assume_regular_freq
     prefer_fast = len(frequency) > 200 and (
         assume_regular_frequency or _is_regular(frequency)
     )
-    prefer_scipy = "scipy" in methods and dy is None and not fit_mean
+    prefer_scipy = "scipy" in methods and dy is None and not (fit_mean and SCIPY_LT_1_15)
```

Comment:
This logic is off I think – e.g. if `fit_mean=True` and scipy is version 1.14, this will return True, and then further down this will lead to an error because scipy 1.14 does not support `fit_mean=True`.

Reply:
But `not ()` is `False`, so this is correctly skipped with 1.14 (those tests are passing, but I think I need to rebase to get rid of the failure in `modeling`).